### PR TITLE
Added support for Keysight P5003A VNA with Spectrum Analysis option

### PR DESCRIPTION
**Pull request by wnluke**
Spectrum Analysis is enabled during the init.
Since reset command switches to VNA option, reset command is implemented in sub-class and includes Spectrum Analysis enable

### CODE REVIEW

File: pymeasure/instruments/keysight/keysightP5003A.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            "Keysight P5003A Spectrum Analyzer",
+            **kwargs
+        )
+        """ Enable Spectrum Analysis """
+        self.write("CALCulate1:MEASure1:DELete")
+        self.write("CALCulate:MEASure:DEFine 'a1:Spectrum Analyzer'")
+        self.write("DISP:MEAS:FEED 1")
+
+    input_attenuation = Instrument.control(
+        "SOURce:POWer:ATTenuation:RECeiver:TEST?;", "SOURce:POWer:ATTenuation:RECeiver:TEST %d;",
```
Comment (msmttchr) [inline pymeasure/instruments/keysight/keysightP5003A.py:62]: All this property can be replaced by

```
  input_attenuation_get_command =  "SOURce:POWer:ATTenuation:RECeiver:TEST?
  input_attenuation_set_command = "SOURce:POWer:ATTenuation:RECeiver:TEST %d;"
```
Similar comment for other properties

File: pymeasure/instruments/keysight/keysightP5003A.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+from io import StringIO
+import numpy as np
+
+
+class KeysightP5003A(SpectrumAnalyzer):
+    """ This class represent a Keysight P5003A VNA with S97090B	Spectrum analysis option  """
+
+    # Customize parameters with values taken from datasheet/user manual 
+
+    reference_level_values = (-500, 500)
```
[the remark this answers is not in the document]
Comment (wnluke) [reply pymeasure/instruments/keysight/keysightP5003A.py:38]: Yes. This is the range you can set

reference_500
reference_m500

File: pymeasure/instruments/keysight/keysightP5003A.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    resolution_bw_values = (5, 3e6)
+
+    input_attenuation_values = (0, 70)
+
+    frequency_points_values = (1, 100003)
+
+    detector_values = ("PEAK", "AVER", "SAMP", "NORM", "NEGP", "PSAM", "PAV")
+
+    def __init__(self, resourceName, **kwargs):
```
[the remark this answers is not in the document]
Comment (wnluke) [reply pymeasure/instruments/keysight/keysightP5003A.py:50]: ok

File: pymeasure/instruments/keysight/keysightP5003A.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (msmttchr) [reply pymeasure/instruments/keysight/keysightP5003A.py:36]: Ok

**Review by msmttchr — APPROVED** [review]
Ok for me. Thanks

**Pull request merged**